### ISSUE TITLE
Accessibility actions labels are not read by VoiceOver

### ISSUE THREAD

**Issue by joonasmkauppinen** [facebook/react-native#53496]
### Description

On iOS the `accessibilityActions` labels are not read out by VoiceOver.

For example, in the code below, VoiceOver only reads the `name` of each accessibility action. Instead it should read the provided `label`.

```js
<Pressable 
  style={styles.button}
  accessibilityActions={[
  { name: 'LIKE', label: 'Add a like.' },
  { name: 'BOOKMARK', label: 'Add item to bookmarks.' },
  { name: 'MENU', label: 'Open menu screen.' },
  ]}
  onAccessibilityAction={event => {
    ...
  }}
>
  <Text style={styles.buttonLabel}>Simple Button</Text>
</Pressable>
```

We noticed this issue after migrating our project to the new a architecture. We have also verified that this **is not** an issue on the old architecture.

### Steps to reproduce

1. Turn on VoiceOver on an iOS device.
2. Move VoiceOver focus on an element that has `accessibilityActions`.
3. Swipe up/down to go through the actions.
4. VoiceOver only reads the action names and not the labels.

### React Native Version

0.79.6

### Affected Platforms

Runtime - iOS

### Areas

Other (please specify)

### Output of `npx @react-native-community/cli info`

```text
Reproducible with the provided Expo app on an iOS device.
```

### Stacktrace or Logs

```text
-
```

### MANDATORY Reproducer

https://github.com/joonasmkauppinen/repro-rn-accessibility-actions-label-issue

**Comment by cortinico** [comment]
So are you saying that `accessibilityActions` is effectively always ignored?

Is this a regression from a previous version of RN or just a missing feature?

**Comment by joonasmkauppinen** [comment]
I wouldn't say `accessibilityActions` are ignored. They are there and users can use them through the screen reader actions menu. Screen reader just announces the wrong property to the user. Screen reader should read the `label`, but it reads  the `name`. This leads to very poor UX since the `name` is just an identifier to distinguish between the different accessibility actions.

Without looking at the native side implementation, it seems that there is just a small bug in the native code where instead of passing the action `label` to the screen reader, the action `name` is passed.

And this does seem to be a regression from an older version of RN. Before we migrated to the new architecture `accessibilityActions` used to work perfectly.
